src/librustc_typeck/astconv.rs

```diff
@@ -1175,14 +1198,13 @@ impl<'o, 'gcx: 'tcx, 'tcx> dyn AstConv<'gcx, 'tcx> + 'o {
                 } else {
                     self.re_infer(span, None).unwrap_or_else(|| {
                         span_err!(tcx.sess, span, E0228,
-                                  "the lifetime bound for this object type cannot be deduced \
-                                   from context; please supply an explicit bound");
+                            "the lifetime bound for this object type cannot be deduced \
```

Comment:
```suggestion
                            "the lifetime bound for this object type cannot be inferred \
```
while reading this PR again, I noticed this... we use "infer" elsewhere and so we should prefer that everywhere instead of C++ism like "deduce".

Reply:
That should be a separate PR I think. You're welcome to make it though, and I can review. ;-)

Reply:
@alexreg Sure; leaving it here (unresolved) for visibility and as a reminder ;)

Reply:
@centril With this PR merging now, just a note for you that this is unresolved, in case you want to create an issue or PR for it yourself.